qiskit/circuit/random/utils.py

```diff
@@ -11,16 +11,385 @@
+
+    # Now, zeros are filtered out in above if-block.
+    # Now, If none of the edges_probs are `None`
+    if all(edges_probs):
+        if not np.isclose(np.sum(edges_probs), 1.000, rtol=0.001):
+            # Normalize edge weights if not already normalized.
+            edges_probs = edges_probs / np.sum(edges_probs)
+
+    # If any of the values of the probability is None, then it would raise an error.
+    elif any(edges_probs):
+        raise ValueError(
+            "Some of the probabilities of a qubit-pair getting selected is `None`"
+            " It should either be all `None` or all positive numerical weights. "
+        )
+
+    # If all edge weights are none, assume the weight of each edge to be 1/N.
+    elif None in edges_probs:
+        edges_probs = [1.0 / num_edges for _ in range(num_edges)]
+
+    # edge weights in interaction_graph must be positive
+    for prob in edges_probs:
+        if prob < 0:
+            raise ValueError("The probability should be positive")
+
+    gates_1q = None
+    if insert_1q_oper:
+        gates_1q = _get_gates(n_qubits=1)
+
+        # Remove `measure` and `reset` from 1Q operations.
+        gates_1q = [
+            (instr, num_gates, num_params)
+            for instr, num_gates, num_params in gates_1q
+            if not instr.name in {"measure", "reset"}
+        ]
+        gates_1q = np.array(
+            gates_1q, dtype=[("class", object), ("num_qubits", np.int64), ("num_params", np.int64)]
+        )
+
+    gates_2q = np.array(
+        _get_gates(n_qubits=2),
+        dtype=[("class", object), ("num_qubits", np.int64), ("num_params", np.int64)],
+    )
+
+    qc = QuantumCircuit(num_qubits)
+
+    if measure or conditional:
+        cr = ClassicalRegister(num_qubits, "c")
+        qc.add_register(cr)
+
+    if seed is None:
+        seed = np.random.randint(0, np.iinfo(np.int32).max)
+    rng = np.random.default_rng(seed)
+
+    qubits = np.array(qc.qubits, dtype=object, copy=True)
+
+    edges_used = {edge: 0 for edge in edge_list}
+
+    # Declearing variables, so that lint doesn't complaint.
+    reset_2q = None
+    cond_val_2q = None
+    cond_val_1q = None
+
+    # If conditional is not required, there is not need to calculate random numbers.
+    # But, since the variables are required in the for-loop so let's get an array of false.
+    if not conditional:
+        cond_1q = np.zeros(num_qubits, dtype=bool)
+        cond_2q = np.zeros(num_edges, dtype=bool)
+
+    # Similarly, if resets are not required, then since, the variable is required
+    # in the for-loop, let's get an array of false.
+    if not reset:
+        reset_2q = np.zeros(num_edges, dtype=bool)
+
+    # This loop will keep on applying gates to qubits until every qubit-pair
+    # has 2Q operations applied at-least `min_2q_gate_per_edge` times.
+    while not all(np.array(list(edges_used.values())) >= min_2q_gate_per_edge):
```

Comment:
I'm wondering, do we want to exclude an edge from being assigned a random 2q gate once the `min_2q_gate_per_edge` constraint is satisfied for it? Maybe as an option? Not sure what is the required behavior here, so this can be ignored.